evm_loader/lib/src/solana_simulator/mod.rs

```diff
@@ -34,16 +35,21 @@ pub struct SolanaSimulator {
 
 impl SolanaSimulator {
     pub async fn new(rpc: &impl Rpc) -> Result<Self, Error> {
-        Self::new_with_config(rpc, RuntimeConfig::default()).await
+        Self::new_with_config(rpc, RuntimeConfig::default(), SyncState::Yes).await
+    }
+
+    pub async fn new_without_sync(rpc: &impl Rpc) -> Result<Self, Error> {
+        Self::new_with_config(rpc, RuntimeConfig::default(), SyncState::No).await
     }
 
     pub async fn new_with_config(
```

Comment:
This method is public and used 3 times: once in the 2 builder methods above and third time in `simulate_solana` module with `SyncState::Yes`.

Each of the 2 builder methods above (`SolanaSimulator::new` and `SolanaSimulator::new_without_sync`) is only used once:
- in `EmulatorAccountStorage::execute_external_instruction` with `SyncState::Yes`;
- in `CallDbClient::build_config_simulator` with `SyncState::No`.

It can be a bit difficult to understand where `SyncState` is `Yes` or `No`. I would suggest either:
- inlining the two `SolanaSimulator::new` and `SolanaSimulator::new_without_sync` helper methods **or**
- making `new_with_config` private and add `RuntimeConfig` as a parameter for `SolanaSimulator::new` method.